### PR TITLE
CHANGELOG.md: Update for 23.11rc1

### PR DESCRIPTION
**Pull request by nepet**
Update changelog, pyln versions, clnrest version and poetry.lock.

I will add another PR adding clnrest and poetry.lock to the release checklist.

### CODE REVIEW

**Comment by nepet** [comment]
Rebased to include recent changes

**Comment by nepet** [comment]
@endothermicdev mentioned that we may order the changelog by the level of interest. If this seems complete I'll first proceed with a rc1 and change the order for the final release!

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,133 @@ All notable changes to this project will be documented in this file.
...
+ - Config: `invoices-onchain-fallback` to automatically add an onchain p2tr address to invoices, and allow that for payment. ([#6421])
+ - hsmd protocol: Added hsmd_check_outpoint and hsmd_lock_outpoint ([#6760])
+ - JSON-RPC: `recover` command to force (unused) lightningd node to restart with `--recover` flag. ([#6772])
+ - Config: `--recover` can take a 32-byte hex string, as well as codex32. ([#6772])
+ - JSON-RPC: `check` now does much more checking on every command (not just basic parameter types). ([#6772])
+ - JSON-RPC: `datastoreusage`: returns the total bytes that are stored under a given key. ([#6442])
+ - Plugins: plugins can now specify (unknown) even messages we should accept from peers. ([#6689])
+ - New `addpsbtoutput` command for creating a PSBT that can receive funds to the on-chain wallet. ([#6676])
+ - Config: `--developer` enables developer options and changes default to be "disable deprecated APIs". ([#6311])
+ - runes: `per=Nsec/min/hour/msec/usec/nsec` for general ratelimiting ([#6617])
```
Comment (endothermicdev) [inline CHANGELOG.md:32]: Would be nice to unify the formatting on all entries where feasible.
```suggestion
 - Runes: `per=Nsec/min/hour/msec/usec/nsec` for general ratelimiting. ([#6617])
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,133 @@ All notable changes to this project will be documented in this file.
...
+ - JSON-RPC: `getroute` now documents that it ignores `fuzzpercent`. ([#6697])
+ - rune: use runes table `id` instead `runes_uniqueid` from `vars` because it returns incorrect unique id if rune/s migrated from datastore. ([#6715])
+ - Added docs, testing, and some fixes related to splicing out, insufficent balance handling, and restarting during a splice. ([#6677])
+
+
+### EXPERIMENTAL
+
+ - Fixed anchor spending to be able to use more than one UTXO. ([#6780])
+ - JSON-RPC: added new dual-funding state `DUALOPEND_OPEN_COMMITTED` ([#6628])
+
```
Comment (endothermicdev) [inline CHANGELOG.md:82]: I think #6752 might deserve a mention.  I didn't notice any others missing, but didn't do an exhaustive search.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,133 @@ All notable changes to this project will be documented in this file.
...
+
+[#6628]: https://github.com/ElementsProject/lightning/pull/6628
+[#6753]: https://github.com/ElementsProject/lightning/pull/6753
+[#6753]: https://github.com/ElementsProject/lightning/pull/6753
```
Comment (endothermicdev) [inline CHANGELOG.md:86]: #6753 is linked 6 times!  I think deduplicating the links would keep the changelog more compact.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,116 @@ All notable changes to this project will be documented in this file.
...
+ - Config: `--developer` enables developer options and changes default to be "disable deprecated APIs". ([#6311])
+ - Runes: `per=Nsec/min/hour/msec/usec/nsec` for general ratelimiting ([#6617])
+ - JSON-RPC: `showrunes` new field `last_used` ([#6617])
+ - New configurable Content-Security-Policy (CSP) header for clnrest ([#6686])
+ - New configurable Cross-Origin-Resource-Sharing(CSP) header for clnrest ([#6686])
+
+
+### Changed
+
+ - Spec: dual-funding now follows the next-funding-id rules. ([#6824])
```
Comment (rustyrussell) [inline CHANGELOG.md:40]: This is usually Protocol: BTW

**Comment by nepet** [comment]
I regenerated and ordered a bit by user impact as far as I can tell.
I'll merge this for now to proceed with a release candidate.
Let me know if anything is still missing, I'll add it to the final release then.

**Pull request merged**